src/overlay_ordering.cpp

```diff
@@ -0,0 +1,25 @@
+#include "overlay_ordering.h"
+
+std::map<std::string, int> base_mutation_overlay_ordering;
+std::map<std::string, int> tileset_mutation_overlay_ordering;
+
+void load_overlay_ordering( JsonObject &jsobj )
```

Comment:
`load_overlay_ordering` is called from init.cpp when a world is loaded. This can happen several times (e.g. when the player saves/dies and starts a new game). The loaded game data can be different for each world (mods) and therefor needs to be reset (see `DynamicDataLoader::unload_data`).

The content of the "overlay_order" is merged into `base_mutation_overlay_ordering` and the map is never cleared. That means a world inherits the overlay ids of *all* the previously loaded worlds.

The map should be cleared from `DynamicDataLoader::unload_data` to avoid this.